internal/parse/parser.go

```diff
@@ -305,23 +305,35 @@ func (p *Parser) parseInputExpression() (*InputPart, bool, error) {
 	return nil, false, nil
 }
 
-// parseStringLiteral parses quoted expressions and ignores their content.
+// parseStringLiteral parses quoted expressions and ignores their content
+// including escaped quotes.
 func (p *Parser) parseStringLiteral() (*BypassPart, bool, error) {
 	cp := p.save()
 
 	if p.pos < len(p.input) {
 		c := p.input[p.pos]
+
 		if c == '"' || c == '\'' {
-			p.skipByte(c)
-			// TODO Handle escaping
-			if !p.skipByteFind(c) {
-				// Reached end of string and didn't find the closing quote.
-				return nil, false, fmt.Errorf("missing right quote in string literal")
+			mark := p.pos
+
+			if p.skipByte(c) {
```

Comment:
Isn't all of the the logic, starting at the very first `if` block, a manual version of `if p.skipByte('"') || p.skipByte('\'')`?

Reply:
You're right, I think I misunderstood the flow of the code when we talked in the call. I've changed it.